### ISSUE TITLE
cannot import @agoric/harden under SES-0.7.4 without noTameRegExp

### ISSUE THREAD

**Issue by warner** [Agoric/SES-shim#237]
The following CJS program:

```js
const { lockdown } = require('ses');
lockdown( { noTameError: true });
require('@agoric/harden'); // fails
```

when run in an initially-empty directory where I've done `yarn add esm ses@0.7.4 @agoric/harden@0.0.4`, yields the following error:

```
$ node -r esm t.js
/tmp/t6/node_modules/@agoric/harden/dist/harden.cjs.js:1
TypeError: RegExp.prototype.source getter called on non-RegExp object
    at Object.get source [as source] (<anonymous>)
    at Object.toString (<anonymous>)
    at /tmp/t6/node_modules/@agoric/harden/dist/harden.cjs.js:280:47
    at Array.forEach (<anonymous>)
    at getAnonIntrinsics (/tmp/t6/node_modules/@agoric/harden/dist/harden.cjs.js:266:20)
    at buildTable (/tmp/t6/node_modules/@agoric/harden/dist/harden.cjs.js:1410:22)
    at Object.<anonymous> (/tmp/t6/node_modules/@agoric/harden/dist/harden.cjs.js:1430:22)
    at Object.<anonymous> (/tmp/t6/node_modules/@agoric/harden/dist/harden.cjs.js:1)
```

The line in question (`harden.cjs.js` line 280) is trying to `toString` an undeniable:

```js

  const undeniableTuples = [
    ['Object.prototype', Object.prototype, {}],
    ['Function.prototype', Function.prototype, function foo() {}],
    ['Array.prototype', Array.prototype, []],
    ['RegExp.prototype', RegExp.prototype, /x/],
    ['Boolean.prototype', Boolean.prototype, true],
    ['Number.prototype', Number.prototype, 1],
    ['String.prototype', String.prototype, 'x'],
    ['%Generator%', Generator, aStrictGenerator],
    ['%AsyncGenerator%', AsyncGenerator, aStrictAsyncGenerator],
    ['%AsyncFunction%', AsyncFunctionPrototype, aStrictAsyncFunction],
  ];

  undeniableTuples.forEach(tuple => {
    const name = tuple[0];
    const undeniable = tuple[1];
    let start = tuple[2];
    if (start === undefined) {
      return;
    }
    start = Object(start);
    if (undeniable === start) {
      return;
    }
    if (undeniable === getProto(start)) {
      return;
    }
    throw new Error(`Unexpected undeniable: ${undeniable}`);
  });
```

If I change the `${undeniable}` to `${name}`, it reveals the unexpected undeniable:

```
Error: Unexpected undeniable: RegExp.prototype
```

I'm guessing that our fix for #230 was incomplete somehow, and the check that `harden` does is confused. There are two problems here: the first triggering this Error, and the second is that the undeniable in question throws a second error when the first one tries to call `toString` on `RegExp.prototype`.

**Comment by warner** [comment]
Oh and of course the problem goes away if I add `noTameRegExp: true` to the `lockdown()` options, as you'd expect.

**Comment by erights** [comment]
Fixed in #243
